partner_event/models/res_partner.py

```diff
@@ -54,3 +55,11 @@ def write(self, data):
         res = super(ResPartner, self).write(data)
         self.mapped('registrations').partner_data_update(data)
         return res
+
+    @api.multi
+    def unlink(self):
+        if self.registration_count > 0:
```

Comment:
No need of `> 0`

Reply:
But I think you can just put on the one2many ondelete='restrict'. I'm not sure if it's going to work, but you have tests for that.